server/pbench/bin/pbench-package-tarballs

```diff
@@ -0,0 +1,85 @@
+#! /bin/bash
+
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+remotearchive="/pbench/archive/fs-version-001"
+
+# TOP, TMP, LOGSDIR, ARCHIVE, TS are all defined in pbench-base.sh
+. $dir/pbench-base.sh
+if [[ -z "$mail_recipients" ]] ;then
+    echo "$PROG: mail_recipients is not defined"
+    exit -1
+fi
+
+files=$(if cd $remotearchive;then find . -path '*/TO-SYNC/*.tar.xz' -printf '%P\n' ;else exit 1 ;fi)
+
+# check if we are running under the unit test regime
+if [ "${_PBENCH_SERVER_TEST}" = "1" ] ;then
+    # make the names reproducible for unit tests
+    tmp=$TMP/$PROG.XXXXX
+    status=status.XXXXX
+    # don't submit any tarballs for further processing
+    dryrun=--dry-run
+else
+    tmp=$TMP/$PROG.$$
+    status=status.$$
+    dryrun=
+fi
+
+trap "rm -rf $tmp" EXIT
+
+mkdir -p "$tmp"
+
+#log_init $(basename $0)
+logdir=$LOGSDIR/$(basename $0)/$prefix/$TS
+mkdir -p "$logdir"
+rc=$?
+if [[ $rc != 0 ]] ;then
+    echo Failed: mkdir -p "$logdir"
+    exit -2
+fi
+
+calculate_prefixname () {
+    tarname=${1##*/}
+    prefixname=${tarname%%.tar.xz}
+    despath=${1%/*}
+    prefixpath=`echo $prefixname | sed 's;^;'.prefix/prefix.';'`
+    prefix="$despath/$prefixpath"
+}
+
+calculate_md5_prefix () {
+    for tar in ${tar_path[@]}
+    do
+        echo "$tar" >> $tmp/file-to-sync
+	echo "$tar.md5" >> $tmp/file-to-sync
+	calculate_prefixname $tar
+        if [ -s $remotearchive/$prefix ]; then
+	    echo "$prefix" >> $tmp/file-to-sync
+        fi
+    done
+}
+
+tar_path=$(echo $files | sed 's/\/TO-SYNC//g')
+calculate_md5_prefix $tar_path
+
+pushd $remotearchive >/dev/null || exit 2
+if [ ! -d $remotearchive ] ;then
+    mkdir -p $remotearchive
+fi
+
+sync_content=$(cat $tmp/file-to-sync)
+tar --create --file $tmp/rsync-blob.tar $sync_content
```

Comment:
Better to use  the -T (--files-from) option to tar: avoids long command line problems.
"file-to-sync" should probably be called "files-to-sync".